daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -779,27 +780,59 @@ private[lf] final class Compiler(
     }
   }
 
+  private[this] def translateTemplatePreConditionCheck(
+      env: Env,
+      templateId: Identifier,
+      contract: s.SExpr,
+      body: Env => s.SExpr,
```

Comment:
Need to have an `Env` dependence as we are wrapping the body in let's and this will alter the environment